src/main/java/Note/ui/Storage.java

```diff
@@ -0,0 +1,107 @@
+package Note.ui;
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Handles saving and loading of tasks to and from a file.
+ */
+public class Storage {
+
+    private String filePath;
+
+    /**
+     * Constructs a Storage object for the given file path.
+     *
+     * @param filePath the path to the file where tasks are stored
+     */
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    /**
+     * Saves a list of tasks to the file.
+     * Each task is stored in a line in the format:
+     * <Type> | <Done status> | <Description> | <Additional info>
+     *
+     * @param tasks the list of tasks to save
+     * @throws IOException if an I/O error occurs while writing to the file
+     */
+    public void save(List<Task> tasks) throws IOException {
+        File folder = new File(filePath).getParentFile();
+        if (!folder.exists()) {
+            folder.mkdirs(); // create folder if it doesn't exist
+        }
+
+        try (BufferedWriter writer = new BufferedWriter(new FileWriter(filePath))) {
+            for (Task t : tasks) {
+                String line = "";
+                switch (t.getTypeIcon()) { // use getTypeIcon() instead of getType()
+                    case "T":
```

Comment:
It would be great if you could match the indentation of the `switch` and `case` staements here to align with the coding standard.
```suggestion
                    switch (t.getTypeIcon()) { // use getTypeIcon() instead of getType()
                    case "T":
```